src/NotificationServices/Softeq.NetKit.Services.SmsNotifications/SmsSender/TwilioSmsSender.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Developed by Softeq Development Corporation
+// http://www.softeq.com
+
+using Softeq.NetKit.Services.SmsNotifications.Exception;
+using Softeq.NetKit.Services.SmsNotifications.Models;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Twilio;
+using Twilio.Rest.Api.V2010.Account;
+
+namespace Softeq.NetKit.Services.SmsNotifications.SmsSender
+{
+    public class TwilioSmsSender : ISmsSender
+    {
+        private readonly TwilioSmsConfiguration _twilioSmsConfiguration;
+        public TwilioSmsSender(TwilioSmsConfiguration twilioSmsConfiguration)
+        {
+            _twilioSmsConfiguration = twilioSmsConfiguration;
+            InitTwilioClient();
+        }
+
+        private void InitTwilioClient()
+        {
+            if (string.IsNullOrEmpty(_twilioSmsConfiguration.AccountSid) || string.IsNullOrEmpty(_twilioSmsConfiguration.AuthToken) || string.IsNullOrEmpty(_twilioSmsConfiguration.FromNumber))
+            {
+                throw new SmsSenderException($"Error while sending sms. Config file does not contain one or more parameters for twilio client!");
+            }
+
+            TwilioClient.Init(_twilioSmsConfiguration.AccountSid, _twilioSmsConfiguration.AuthToken);
```

Comment:
Could you please check how it should be registered. like singleton or nor? It seems to me that we shouldn't Init client in ctor

Reply:
fixed. Created separate class TwilioHttpClient to avoid initializing TwilioClient in smssender class